src/audio/dai.c

```diff
@@ -747,6 +748,11 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 		 */
 		dconfig->frame_fmt = SOF_IPC_FRAME_S32_LE;
 
+		 /* Set burst length to align with DMAT value in the
+		  * Audio Link Hub.
+		  */
+		dd->config.burst_elems = ALH_GPDMA_BURST_LENGTH;
+
```

Comment:
could we make this configurable over IPC?
This value could be the default but if the driver tries something else the firmware would adjust. we have plenty of spare fields:

````c
/* ALH Configuration Request - SOF_IPC_DAI_ALH_CONFIG */
struct sof_ipc_dai_alh_params {
	struct sof_ipc_hdr hdr;
	uint32_t stream_id;

	/* reserved for future use */
	uint32_t reserved[15];
} __packed;
````

Reply:
Isn't this too tighly coupled to the HW DMA/DAI to be configurable or do you expect to see this vary with use case or hw_params ?

Reply:
currently the value is constant at the driver level and programmed at the hw-params stage.
I think it's a bit risky if there is a constant on both sides.
We don't have this problem with SSP since the firmware programs the DAI and DMA parts, but here you have a firmware and kernel driver programming the two parts independently. So it'd be wise to pass whatever the kernel used and make sure the firmware adjusts.

Reply:
ok, fine by me. We maybe want to log this in dma start() if driver changes FW default, that way it would be obvious if the user had bad values.

Reply:
never mind, we should do this configuration at a later point.

The structure needed to pass this information between the SoundWire and SOF parts would require a change to convey the DMAT value, and I don't want to do this for now (the SoundWire reviews are longer than I expected).

So if Rander confirms this works on ICL, we can take this code and do the configuration later. I don't want this paranoia check to delay the SoundWire baseline further.